### PR TITLE
feat: add alarms for attachments

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Add 2 new alarms related to attachments:

- watch the S3 bucket size holding file attachments. Not sure about threshold but planned to have a warning alarm at 500 MB on staging and 100 GB in production, as a start
- add a warning alarm when we have more than 300 HTTP 4XX requests on the document download API. 4XX requests will happen when requests are malformed or when documents are not found on S3, may be a sign that people try to guess private links

Trello: https://trello.com/c/eLN4TwTf/402-add-alarms-related-to-attachments

### CODE REVIEW

**Comment by AntoineAugusti** [comment]
Will update the spreadsheet as soon as we reach an agreement on alarms

File: aws/common/cloudwatch_alarms.tf [hunk 1]
```diff
@@ -422,3 +422,20 @@ resource "aws_cloudwatch_metric_alarm" "contact-3-500-error-15-minutes-critical"
...
+resource "aws_cloudwatch_metric_alarm" "document-download-bucket-size-warning" {
+  alarm_name          = "document-download-bucket-size-warning"
+  alarm_description   = "Document download S3 bucket size is larger than ${var.alarm_warning_document_download_bucket_size_gb} GB"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = "BucketSizeBytes"
+  namespace           = "AWS/S3"
+  period              = 60 * 15
+  statistic           = "Average"
+  threshold           = var.alarm_warning_document_download_bucket_size_gb * 1000 * 1000 * 1000 # Convert the GB variable to bytes
```
[the remark this answers is not in the document]
Comment (AntoineAugusti) [reply aws/common/cloudwatch_alarms.tf:435]: https://docs.ukcloud.com/articles/other/other-ref-gib.html

GB = Gigabyte powers of 1000
GiB = Gibibyte powers of 1024

I'm more used to GB but happy to change. S3 gives the metric in bytes.

The [pricing page](https://aws.amazon.com/s3/pricing/) mentions

> Amazon S3 storage usage is calculated in binary gigabytes (GB), where 1GB is 2^30 bytes. This unit of measurement is also known as a gibibyte (GiB), defined by the International Electrotechnical Commission (IEC). Similarly, 1TB is 2^40 bytes, i.e. 1024 GBs.

So it seems like they GB as GiB 🤷

[the remark this answers is not in the document]
Comment (AntoineAugusti) [reply aws/common/cloudwatch_alarms.tf]: Changed in the next commit

File: aws/eks/cloudwatch_alarms.tf [hunk 2]
```diff
@@ -75,6 +75,24 @@ resource "aws_cloudwatch_metric_alarm" "load-balancer-10-502-error-5-minutes-cri
   }
 }
 
+resource "aws_cloudwatch_metric_alarm" "document-download-api-high-request-count-warning" {
+  alarm_name          = "document-download-api-high-request-count-warning"
+  alarm_description   = "More than 300 4XX requests in 10 minutes on ${aws_alb_target_group.notification-canada-ca-document-api.name} target group"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "1"
+  metric_name         = "HTTPCode_Target_4XX_Count"
```
Comment (QuanNguyen-CDS) [inline aws/eks/cloudwatch_alarms.tf:83]: is there any benefit in alarming on to 5xx errors as well?

**Review by QuanNguyen-CDS — APPROVED** [review]
LGTM

Note: I tried to understand if threshold (which i think is a float) can overflow or lose resolution. I couldn't find any documentation online that suggested that this would be the case.

**Pull request merged**